packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/TranscribedLines/TranscribedLines.js

```diff
@@ -17,24 +18,47 @@ export const ConsensusLine = styled('g')`
   }
 `
 
-function TranscribedLines ({ lines, scale, task, theme }) {
-  function createMark (line) {
-    const { activeTool, activeToolIndex, setActiveMark } = task
+function TranscribedLines ({ lines, onFinish, scale, task, theme }) {
+  const ref = createRef()
+  const {
+    activeTool,
+    activeToolIndex,
+    setActiveMark
+  } = task
+
+  function onFinishWithRef(event) {
+    onFinish(event, ref.current)
+  }
+
+  function createMark (event, line) {
     const [{ x: x1, y: y1 }, { x: x2, y: y2 }] = line.points
     const markSnapshot = { x1, y1, x2, y2, toolIndex: activeToolIndex }
 
     if (activeTool) {
       const mark = activeTool.createMark(markSnapshot)
       mark.finish()
       setActiveMark(mark)
+
+      let previousAnnotations = []
+      activeTool.tasks.forEach((task) => {
+        const previousAnnotation = {
+          id: mark.id,
```

Comment:
Are identifiers required to be unique for each of these annotations?

Reply:
Yes, I think so. I think the variable naming here actually is adding to the confusion, so I'll think about better naming for this and the associated model.

Reply:
Actually, I switched up the identifier to be the task key because this is already a child node of the mark which is uniquely identified. I do need an identifier, but it can just be the task key which matches the subtask.